### PR TITLE
Permission switch

### LINKED ISSUE #22: Replace the stupid member system with Django's premission system

### PR DESCRIPTION
**Pull request by dandeduck**
Added guardian, and used its instance based permissions. A quick and easy PR. (I decide what issue to work on randomly)

### CODE REVIEW

**Comment by dandeduck** [comment]
I didn't do too much testing, but from what I can tell it works fine

**Comment by dandeduck** [comment]
In order to test this, you can follow [this](https://django-guardian.readthedocs.io/en/stable/userguide/assign.html#assign-obj-perms) and use the view_partner permission (since that's what I used

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 django==2.2.12
-uuid
\ No newline at end of file
+uuid
+guardian
```
Comment (tomtzook) [inline requirements.txt]: It's `django-guardian`

File: pages/views.py [hunk 2]
```diff
@@ -27,10 +27,12 @@ def login_view(request):
...
-    if has_group(user):
-        if is_staff(user):
-            return redirect('/staff/')
-        return redirect('/partners/'+first_group_name(user))
+    partners = user_partners(user)
+
+    if partners:
+        if len(partners) > 1:
+            return redirect('/partners/')
+        return redirect('/partner/'+partners[0])
```
Comment (tomtzook) [inline pages/views.py]: `partners[0].name`

File: package-tracking/urls.py [hunk 3]
```diff
@@ -20,10 +20,10 @@
...
-    path('partners/<str:partner_name>/', partner_view),
-    path('partners/<str:partner_name>/<str:order_id>/', order_edit_view),
-    path('packages/<str:package_id>/', package_view),
-    path('orders/<str:order_id>/', order_view),
-    path('staff/', staff_view),
+    path('partner/<str:partner_name>/', partner_view),
+    path('partner/<str:partner_name>/<str:order_id>/', order_edit_view),
+    path('package/<str:package_id>/', package_view),
+    path('order/<str:order_id>/', order_view),
+    path('partners/', partners_view),
```
Comment (tomtzook) [inline package-tracking/urls.py]: You can also do `partner/`, showing an hierarchal relationship with the other partner paths.

**Review by tomtzook — CHANGES_REQUESTED** [review]
You should add the option to modify permissions from the admin page.

**Comment by tomtzook** [comment]
Forgot to close... oops

**Comment by dandeduck** [comment]
> Forgot to close... oops

Maan I am playing too much Horizon... BUT WHAT JS HORIZON? AM I HER CLONE? WHY IS THE VP OF SALES SUCH A DOUCHE?

File: requirements.txt (hunk 1 above)
Comment (dandeduck) [reply requirements.txt]: oopsy

File: pages/views.py (hunk 2 above)
Comment (dandeduck) [reply pages/views.py]: yup

File: package-tracking/urls.py (hunk 3 above)
Comment (dandeduck) [reply package-tracking/urls.py]: good idea

Comment (tomtzook) [reply package-tracking/urls.py]: It can be `partner/` and that's it. Not need for `partner/partners`

**Comment by tomtzook** [comment]
One small thing

File: package-tracking/urls.py (hunk 3 above)
Comment (dandeduck) [reply package-tracking/urls.py]: oh, without the id?

Comment (dandeduck) [reply package-tracking/urls.py]: ok changed it

**Comment by dandeduck** [comment]
Changed it

File: package-tracking/urls.py (hunk 3 above)
Comment (tomtzook) [reply package-tracking/urls.py]: Did you? Because it doesn't seem like you pushed it.

**Comment by tomtzook** [comment]
You didn't push!

**Comment by dandeduck** [comment]
> You didn't push!

fuck frocking password protected ssh key... I am away from home, will push later tonight

**Comment by dandeduck** [comment]
pushed it finally fuckkkk

**Review by tomtzook — APPROVED** [review]

**Comment by tomtzook** [comment]
Merge when wanted

**Pull request merged**